backend/src/main/java/ca/bc/gov/hlth/hnweb/controller/EligibilityController.java

```diff
@@ -54,23 +53,31 @@
 
 	private static final Logger logger = LoggerFactory.getLogger(EligibilityController.class);
 
+	@Deprecated
 	private static final String DATE_FORMAT_yyyyMMdd = "yyyyMMdd";
 
 	@Autowired
 	private EligibilityService eligibilityService;
 	
-	@GetMapping("/check-eligibility")
-	public ResponseEntity<CheckEligibilityResponse> checkEligibility(@RequestParam(name = "phn", required = true) String phn,
-			@DateTimeFormat(iso = ISO.DATE) @RequestParam(name = "eligibilityDate", required = false) Date eligibilityDate) {
-		logger.info("checkEligibility request - phn: {} date: {}", phn, eligibilityDate);
+	@Autowired
+	private MSHDefaults mshDefaults;
+	
+	@PostMapping("/check-eligibility")
+	public ResponseEntity<CheckEligibilityResponse> checkEligibility(@Valid @RequestBody CheckEligibilityRequest checkEligibilityRequest) {
 
 		try {
-			CheckEligibilityResponse checkEligibilityResponse = eligibilityService.checkEligibility(phn, eligibilityDate);			
-			ResponseEntity<CheckEligibilityResponse> response = ResponseEntity.ok(checkEligibilityResponse);		
+			R15Converter converter = new R15Converter(mshDefaults);
+			R15 r15 = converter.convertRequest(checkEligibilityRequest);
+			Message message = eligibilityService.checkEligibility(r15);
```

Comment:
This Message object is a generic HAPI message type that's just sort of an intermediary that makes it easy to parse the response with the terser right? 
Just for developer readability could we call it r15Response or hapiR15Response or something?

Reply:
Yeah, the generic Message type is fine since we're accessing everything via the Terser and not the object graph. I'll fix the naming.